src/adagrams.js

```diff
@@ -1,8 +1,149 @@
+    // .map copies array for shallow clone 
+    // https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/
+    const copyOfHand = lettersInHand.map((x) => x);
+    // .split is an ordered set of substrings
+    const inputLetters = input.toUpperCase().split('');
+    // check if input is a valid word made from the hand of 10
+    for(let i = 0; i < inputLetters.length; i++) {
+      if(copyOfHand.includes(inputLetters[i])) {
+        // if the letter is in the hand, remove it.
+        copyOfHand.splice(copyOfHand.indexOf(inputLetters[i]), 1);
+      } else {
+        return false;
+      }
+    };
+      return true;
+  },
+
+  // Wave 3
+  // return score of given word  based on point system and additional scoring rules. 
+  scoreWord(word) {
+    let scoreCount = 0;
+    // create array
+    const splitWord = word.toUpperCase().split('');
+    // validate input
+    if (word === "") {
+      return scoreCount;
+    };
+
+    for(let i = 0; i < splitWord.length; i++) {
+      // expression is an Array
+      const expression = splitWord[i];
+      // expression is a String
+      switch (expression) {
+        // specific letters and their values - multiple cases on one line.
+        // https://stackoverflow.com/questions/13207927/switch-statement-multiple-cases-in-javascript
+        case "A": case "E": case "I": case "O": case "U": case "L": case "N": case "R": case "S": case "T":
+          scoreCount += 1;
+          break;
+        case "D": case "G":
+          scoreCount += 2;
+          break;
+        case "B": case "C": case "M": case "P":
+          scoreCount += 3;
+          break;
+        case "F": case "H": case "V": case "W": case "Y":
+          scoreCount += 4;
+          break;
+        case "K":
+          scoreCount += 5;
+          break;
+        case "J": case "X":
+          scoreCount += 8;
+          break;
+        case "Q": case "Z":
+          scoreCount += 10;
+          break;
+      };
+
+    }; 
+    if (splitWord.length >= 7) {
+      scoreCount += 8;
+    };
+    
+    return scoreCount;
 
+  },
+
+
+   highestScoreFrom(words) {
+    //  map returns an array of objects
+    // .map is my favorite! :)
+     const wordScores = words.map((word) => {
+      const newWord = {
+        'word': word,
+        'score': this.scoreWord(word)
+      };
+      return newWord;
+    });
+        
+    // set winningWord to the first object in array as a starting point
+     let winningWord = wordScores[0];
+
+     for (let i in wordScores) {
+       // using dot notation to access the object's properties/values
+      //  check to see if word score is greater and
+      // if so, then assign this new word to winningWord.
+       if (wordScores[i].score > winningWord.score) {
+         winningWord.word = wordScores[i].word;
+         winningWord.score = wordScores[i].score;
+       } else if
+      //  check if two words have same score but one has length 10.
```

Comment:
This is very minor but there is an extra space on the lines in this function. I use a VSCode plugin called indent-rainbow to catch little indentation things like this.